### ISSUE TITLE
Layout/MultilineMethodCallIndentation is broken since 1.84.2

### ISSUE THREAD

**Issue by zverok** [rubocop/rubocop#15120]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Layout/MultilineMethodCallIndentation

### Steps to reproduce the problem

Minimal reproduction:
```ruby
# frozen_string_literal: true

def foo
  {
    val: Table.joins(:other_table)
              .distinct
              .order(:name, :id)
              .map { [some_method(_1), _1.id] },
  }
end
```

Cop settings: 
```
$ bundle exec rubocop --show-cops Layout/MultilineMethodCallIndentation
# Supports --autocorrect
Layout/MultilineMethodCallIndentation:
  Description: Checks indentation of method calls with the dot operator that span more
    than one line.
  Enabled: true
  VersionAdded: '0.49'
  EnforcedStyle: aligned
  SupportedStyles:
  - aligned
  - indented
  - indented_relative_to_receiver
  IndentationWidth: 
```

### Expected behavior

The initial code is considered correct.

### Actual behavior

Registered offense:
```
try_rubocop.rb:6:15: C: [Correctable] Layout/MultilineMethodCallIndentation: Align .distinct with Table.joins(:other_table) on line 5.
              .distinct
              ^^^^^^^^^
```
Suggested autocorrection (sic!):
```ruby
# frozen_string_literal: true

def foo
  {
    val: Table.joins(:other_table)
         .distinct
              .order(:name, :id)
              .map { [some_method(_1), _1.id] },
  }
end
```

This obviously looks wrong to me. Even if we assume that the rule changed from "align with call of `.joins`" to "align with the start of the expression" (though cop's metadata doesn't tell that, saying the cop is added in 0.49, and never modified since), the subsequent methods aren't realigned, so it doesn't produce a visually consistent result.

1.84.0 and below consider the initial code correct.

1.84.1 corrects the code to
```ruby
  {
    val: Table.joins(:other_table)
         .distinct
         .order(:name, :id)
         .map { [some_method(_1), _1.id] },
  }
```
...which is different from what it was before, but at least consistent (though I'd prefer it wouldn't change).

1.84.2 introduces the current behavior. 
If I change it manually to what 1.84.1 did, it considers the code correct, too.

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
$ rubocop -V
1.84.2 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 3.1, running on ruby 3.3.7) [x86_64-linux]

(and any version since that, including the current master)
```

**Comment by zverok** [comment]
Additionally, this reproduces the glitch, too (here, the code is not even "hash value" -- if we'll agree to change of multiline formatting in hash values -- but a body of a block):

```ruby
# frozen_string_literal: true

def foo
  {
    val: -> {
      Table.joins(:other_table)
           .distinct
           .order(:name, :id)
           .map { [some_method(_1), _1.id] }
    }
  }
end
```

Autocorrected to:
```ruby
# frozen_string_literal: true

def foo
  {
    val: -> {
      Table.joins(:other_table)
      .distinct
           .order(:name, :id)
           .map { [some_method(_1), _1.id] }
    }
  }
end
```

**Comment by zverok** [comment]
Actually seems related to or duplicate of https://github.com/rubocop/rubocop/issues/15054

**Comment by fynsta** [comment]
Does not reproduce on master anymore, I think #15054 fixed it.
